superset/explore/api.py

```diff
@@ -0,0 +1,126 @@
+    resource_name = "explore"
+    openapi_spec_tag = "Explore"
+    openapi_spec_component_schemas = (ExploreContextSchema,)
+
+    @expose("/", methods=["GET"])
+    @protect()
+    @safe
+    @event_logger.log_this_with_context(
+        action=lambda self, *args, **kwargs: f"{self.__class__.__name__}.get",
+        log_to_statsd=True,
+    )
+    def get(self) -> Response:
+        """Retrives Explore initial context.
+        ---
+        get:
+          description: >-
+            Retrives Explore initial context.
+          parameters:
+          - in: query
+            schema:
+              type: string
+            name: form_data_key
+          - in: query
+            schema:
+              type: string
+            name: permalink_key
+          - in: query
+            schema:
+              type: integer
+            name: slice_id
+          - in: query
+            schema:
+              type: integer
+            name: dataset_id
+          - in: query
+            schema:
+              type: string
+            name: dataset_type
+          responses:
+            200:
+              description: Returns the initial context.
+              content:
+                application/json:
+                  schema:
+                    $ref: '#/components/schemas/ExploreContextSchema'
+            400:
+              $ref: '#/components/responses/400'
+            401:
+              $ref: '#/components/responses/401'
+            404:
+              $ref: '#/components/responses/404'
+            422:
+              $ref: '#/components/responses/422'
+            500:
+              $ref: '#/components/responses/500'
+        """
+        try:
+            params = CommandParameters(
+                actor=g.user,
+                permalink_key=request.args.get("permalink_key"),
+                form_data_key=request.args.get("form_data_key"),
+                dataset_id=request.args.get("dataset_id"),
+                dataset_type=request.args.get("dataset_type"),
+                slice_id=request.args.get("slice_id"),
+            )
+            result = GetExploreCommand(params).run()
+            if not result:
+                return self.response_404()
+            return self.response(200, result=result)
+        except DatasetAccessDeniedError as ex:
+            return self.response(
+                403,
+                message=ex.message,
+                dataset_id=ex.dataset_id,
+                dataset_type=ex.dataset_type,
+            )
+        except (ChartNotFoundError, ExplorePermalinkGetFailedError) as ex:
+            return self.response(404, message=str(ex))
+        except WrongEndpointError as ex:
+            return self.response(302, redirect=ex.redirect)
+        except TemporaryCacheAccessDeniedError as ex:
+            return self.response(403, message=str(ex))
+        except TemporaryCacheResourceNotFoundError as ex:
+            return self.response(404, message=str(ex))
```

Comment:
I really really believe these common errors should be handled in some kind of base class. We shouldn't have to duplicate the same code in every API endpoint.

Reply:
I agree. We even created a task in our backlog to do this work and adjust all the endpoints but it always loses the priority battle 😞. Do you think you can work on it?